incorrect calling convention for exported variadic functions

### Zig Version

0.14.0-dev.2034+56996a280

### Steps to Reproduce and Observed Behavior

```sh
$ cat callconv.zig
const std = @import("std");

export fn print(msg: [*:0]const u8, ...) void {
    std.debug.print("{s}\n", .{msg});
}

$ zig build-lib callconv.zig -target aarch64-linux
callconv.zig:1:1: error: calling convention 'x86_64_sysv' only available on architectures 'x86_64'
const std = @import("std");
^~~~~
referenced by:
    root: /root/lang/zig/ziglang/lib/std/start.zig:3:22
    comptime: /root/lang/zig/ziglang/lib/std/start.zig:28:9
    2 reference(s) hidden; use '-freference-trace=4' to see all references
```

The compiler fails to infer the correct calling convention for exported functions that accept variable arguments.

But if we manually specify the calling convention by adding `callconv(.c)` to the function declaration, the function compiles without any issues. For instance:
```zig
export fn print(msg: [*:0]const u8, ...) callconv(.c) void {
    std.debug.print("{s}\n", .{msg});
}
```

Also, the error message is vague and does not specify which line of the code is problematic.

### Expected Behavior

The compiler should correctly infer the calling convention without the need for manual intervention.

--- a/src/Compilation.zig
+++ b/src/Compilation.zig
@@ -1962,6 +1962,7 @@ pub fn destroy(comp: *Compilation) void {
     comp.failed_win32_resources.deinit(gpa);
 
     comp.link_diags.deinit();
+    comp.link_task_queue.deinit(gpa);
 
     comp.clearMiscFailures();
 

--- a/src/Sema.zig
+++ b/src/Sema.zig
@@ -26619,6 +26619,7 @@ fn zirFuncFancy(sema: *Sema, block: *Block, inst: Zir.Inst.Index) CompileError!A
 
     const pt = sema.pt;
     const zcu = pt.zcu;
+    const ip = &zcu.intern_pool;
     const inst_data = sema.code.instructions.items(.data)[@intFromEnum(inst)].pl_node;
     const extra = sema.code.extraData(Zir.Inst.FuncFancy, inst_data.payload_index);
     const target = zcu.getTarget();
@@ -26793,7 +26794,21 @@ fn zirFuncFancy(sema: *Sema, block: *Block, inst: Zir.Inst.Index) CompileError!A
 
             const zir_decl = sema.code.getDeclaration(decl_inst.resolve(&zcu.intern_pool) orelse return error.AnalysisFail)[0];
             if (zir_decl.flags.is_export) {
-                break :cc .C;
+                break :cc target.cCallingConvention() orelse {
+                    // This target has no default C calling convention. We sometimes trigger a similar
+                    // error by trying to evaluate `std.builtin.CallingConvention.c`, so for consistency,
+                    // let's eval that now and just get the transitive error. (It's guaranteed to error
+                    // because it does the exact `cCallingConvention` call we just did.)
+                    const cc_type = try sema.getBuiltinType("CallingConvention");
+                    _ = try sema.namespaceLookupVal(
+                        block,
+                        LazySrcLoc.unneeded,
+                        cc_type.getNamespaceIndex(zcu),
+                        try ip.getOrPutString(sema.gpa, pt.tid, "c", .no_embedded_nulls),
+                    );
+                    // The above should have errored.
+                    @panic("std.builtin is corrupt");
+                };
             }
         }
         break :cc .auto;

